desktop-widgets/diveplanner.cpp

```diff
@@ -195,7 +199,22 @@ void PlannerSettingsWidget::setDiveMode(int mode)
 
 void DivePlannerWidget::setSalinity(int salinity)
 {
-	ui.salinity->setValue(salinity / 10000.0);
+	bool mapped = false;
+	for (int i = 0; i < ui.waterType->count(); i++) {
+		if (salinity == ui.waterType->itemData(0).toInt()) {
```

Comment:
One major issue in the current implementation is that a value of 1.02 is not recognized as "Sea water..." but as "Custom". I first thought this is simply not implemented yet but I think it is but there is an mistake in the code right here...
It has to be `if (salinity == ui.waterType->itemData(i).toInt()) {` ??!
Sorry, no time to test it yet...

Reply:
So true